shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -269,13 +269,23 @@ - (void)popSystemNavigator:(BOOL)isAnimated {
   // It's also possible in an Add2App scenario that the FlutterViewController was presented
   // outside the context of a UINavigationController, and still wants to be popped.
 
-  UIViewController* engineViewController = [_engine.get() viewController];
+  FlutterViewController* engineViewController = [_engine.get() viewController];
   UINavigationController* navigationController = [engineViewController navigationController];
   if (navigationController) {
     [navigationController popViewControllerAnimated:isAnimated];
   } else {
-    UIViewController* rootViewController =
-        [UIApplication sharedApplication].keyWindow.rootViewController;
+    UIViewController* rootViewController = nil;
+#if APPLICATION_EXTENSION_API_ONLY
+    if (@available(iOS 15.0, *)) {
+      rootViewController =
+          [engineViewController windowSceneIfViewLoaded].keyWindow.rootViewController;
+    } else {
+      FML_LOG(WARNING)
+          << "rootViewController is not available in application extension under iOS 15.0.";
```

Comment:
Since it is breaking the pop flow, I imagined that the developer would either stop supporting iOS 15. Or redesign the UI for iOS 15 or lower. 

I think this message is necessary to let the developer know why the pop button doesn't work. It would be hard to figure out if we only introduced a silence no-op.

Reply:
We would probably need to update the documentation for this too. However, I think the warning message is more visible and this one won't be too spammy.